sys/include/phydat.h

```diff
@@ -288,6 +288,35 @@ void phydat_fit(phydat_t *dat, const int32_t *values, unsigned int dim);
  */
 size_t phydat_to_json(const phydat_t *data, size_t dim, char *buf);
 
+/**
+ * @brief   Convert a date and time contained in phydat structs to a Unix timestamp.
+ *          See phydat_unix() for the date notation and peculiarities.
+ *
+ * @param date              Date to use in the timestamp.
+ * @param time              Time to use in the timestamp.
+ * @param offset_seconds    Timezone offset in seconds to use in the timestamp.
+ *
+ * @return  A unix timestamp
+ */
+int64_t phydat_date_time_to_unix(phydat_t *date, phydat_t *time, int32_t offset_seconds);
+
+/**
+ * @brief   Convert a date and time (per ISO8601) to a Unix timestamp (seconds since 1970).
+ *
+ * @param year      Year in the Common Era (CE). Note that 0 is 1 BCE, 1 is 2 BCE, etc.
+ * @param month     Month of the year.
+ * @param day       Day of the month.
+ * @param hour      Hour of the day.
+ * @param minute    Minute of the hour.
+ * @param second    Second of the minute.
+ * @param offset    Timezone offset in seconds.
+ *
+ * @return          A Unix timestamp (seconds since 1970).
+ */
+int64_t phydat_unix(int16_t year, int16_t month, int16_t day,
+                     int16_t hour, int16_t minute, int16_t second,
+                     int32_t offset);
```

Comment:
Isn't this just a re-implementation of `mktime()`?

Reply:
Yes, with a few differences:

- The time offset is explicit instead of implicit.
- It does not have a DST flag.
- It uses `int64_t` instead of `time_t` (`int`) to avoid Y2K38.

I couldn't find any 64-bit `mktime` implementation available in RIOT-OS, but please let me know if there is one.

Reply:
> It uses `int64_t` instead of `time_t` (int) to avoid Y2K38.

Are you sure `time_t` is only 32 bit? For 

```C
printf("time_t is %u bit\n", sizeof(time_t) * 8);
```

I get

    time_t is 64 bit

on both newlib and picolibc.

Reply:
> Are you sure time_t is only 32 bit?

No, this depends on the platform. The ARM-based boards I tested use 64-bit time. ESP32, ATmega328 and *native*, however, give me `time_t is 32 bit`.